app/app.go

```diff
@@ -445,6 +434,38 @@ func New(
 	app.TransferKeeper = uibctransferkeeper.New(ibcTransferKeeper, app.BankKeeper)
 	transferModule := uibctransfer.NewAppModule(ibctransfer.NewAppModule(ibcTransferKeeper), app.TransferKeeper)
 
+	baseBankKeeper := app.BankKeeper.(bankkeeper.BaseKeeper)
+
+	app.bech32IbcKeeper = *bech32ibckeeper.NewKeeper(
+		&app.IBCKeeper.ChannelKeeper, appCodec, keys[bech32ibctypes.StoreKey],
+		&app.TransferKeeper,
+	)
+
+	app.GravityKeeper = gravitykeeper.NewKeeper(
+		keys[gravitytypes.StoreKey],
+		app.GetSubspace(gravitytypes.ModuleName),
+		appCodec,
+		&baseBankKeeper,
+		&stakingKeeper,
+		&app.SlashingKeeper,
+		&app.DistrKeeper,
+		&app.AccountKeeper,
+		&app.TransferKeeper.Keeper,
+		&app.bech32IbcKeeper,
+	)
+
+	// register the staking hooks
+	//
+	// NOTE: The stakingKeeper above is passed by reference, so that it will contain
+	// these hooks.
+	app.StakingKeeper.SetHooks(
+		stakingtypes.NewMultiStakingHooks(
+			app.DistrKeeper.Hooks(),
+			app.SlashingKeeper.Hooks(),
+			app.GravityKeeper.Hooks(),
+		),
+	)
```

Comment:
This way we should be able to get rid of all the `*app.StakingKeeper` lines
```suggestion
	app.StakingKeeper = *stakingKeeper.SetHooks(
		stakingtypes.NewMultiStakingHooks(
			app.DistrKeeper.Hooks(),
			app.SlashingKeeper.Hooks(),
			app.GravityKeeper.Hooks(),
		),
	)
```